src/brpc/policy/http_rpc_protocol.cpp

```diff
@@ -1145,16 +1146,20 @@ ParseResult ParseHttpMessage(butil::IOBuf *source, Socket *socket,
         if (http_imsg->Completed()) {
             CHECK_EQ(http_imsg, socket->release_parsing_context());
             const ParseResult result = MakeMessage(http_imsg);
+            http_imsg->CheckProgressiveRead(arg, socket);
             if (socket->is_read_progressive()) {
                 socket->OnProgressiveReadCompleted();
             }
             return result;
-        } else if (socket->is_read_progressive() &&
-                   http_imsg->stage() >= HTTP_ON_HEADERS_COMPLETE) {
-            // header part of a progressively-read http message is complete,
-            // go on to ProcessHttpXXX w/o waiting for full body.
-            http_imsg->AddOneRefForStage2(); // released when body is fully read
-            return MakeMessage(http_imsg);
+        } else if (http_imsg->stage() >= HTTP_ON_HEADERS_COMPLETE) {
+            http_imsg->CheckProgressiveRead(arg, socket);
```

Comment:
Adding a method lookup in hot path, maybe affect performance

Reply:
OK, I'll make `CheckProgressiveRead` inline.

Reply:
Not inline, I mean the FindMethodPropertyByURI call inside CheckProgressiveRead.

Reply:
Oh I got u. However, one of the keypoints of this feature is to get the corresponding service opt `enable_progressive_read` after header completed and determine the read body behavior. Is there a better way to get this opt without  calling `FindMethodPropertyByURI` to fetch it inside the method property?  Appreciate it.

Reply:
Maybe we can add a flag `has_progressive_read_method` to Server
When Server Start, check all MethodProperty, if any one has progressive_read option, set `has_progressive_read_method` to true
When parsing message, check `has_progressive_read_method`, if true, then call FindMethodPropertyByURI

Reply:
Great idea! Thank u so much.